src/plugins/event-plugins/__tests__/JsErrorPlugin.test.ts

```diff
@@ -422,4 +422,102 @@ describe('JsErrorPlugin tests', () => {
+            Object.assign(promiseRejectionEvent, {
+                promise: new Promise(() => {}),
+                reason: {}
+            })
+        );
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(1);
+        expect(record.mock.calls[0][0]).toEqual(JS_ERROR_EVENT_TYPE);
+        expect(record.mock.calls[0][1]).toMatchObject(
+            expect.objectContaining({
+                version: '1.0.0',
+                type: 'unhandledrejection',
+                message: 'undefined'
+            })
+        );
+    });
+
+    test('when unhandledrejection error event outputs null object as reason then message is recorded as undefined', async () => {
+        // Init
+        const plugin: JsErrorPlugin = new JsErrorPlugin();
+
+        // Run
+        plugin.load(context);
+        const promiseRejectionEvent: PromiseRejectionEvent = new Event(
+            'unhandledrejection'
+        ) as PromiseRejectionEvent;
+        // JSDOM has not implemented PromiseRejectionEvent, so we 'extend'
+        // Event to have the same functionality
+        window.dispatchEvent(
+            Object.assign(promiseRejectionEvent, {
+                promise: new Promise(() => {}),
+                reason: null
+            })
+        );
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(1);
+        expect(record.mock.calls[0][0]).toEqual(JS_ERROR_EVENT_TYPE);
+        expect(record.mock.calls[0][1]).toMatchObject(
+            expect.objectContaining({
+                version: '1.0.0',
+                type: 'unhandledrejection',
+                message: 'undefined'
+            })
+        );
+    });
+
+    test('when unhandledrejection error event outputs error object as reason then error object is used', async () => {
+        // Init
+        const plugin: JsErrorPlugin = new JsErrorPlugin();
+
+        // Run
+        plugin.load(context);
+        const promiseRejectionEvent: PromiseRejectionEvent = new Event(
+            'unhandledrejection'
+        ) as PromiseRejectionEvent;
+        // JSDOM has not implemented PromiseRejectionEvent, so we 'extend'
+        // Event to have the same functionality
+        window.dispatchEvent(
+            Object.assign(promiseRejectionEvent, {
+                promise: new Promise(() => {}),
+                reason: {
+                    name: 'TypeError',
+                    message: 'NetworkError when attempting to fetch resource.',
+                    stack: 't/n.fetch@mock_client.js:2:104522t/n.fetchWrapper'
+                }
+            })
+        );
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(1);
+        expect(record.mock.calls[0][0]).toEqual(JS_ERROR_EVENT_TYPE);
+        expect(record.mock.calls[0][1]).toMatchObject(
+            expect.objectContaining({
+                version: '1.0.0',
+                type: 'unhandledrejection: TypeError',
```

Comment:
I think we should indicate it's from an `unhandledrejection` by appending to the string instead of having to wrap the `JsErrorEvent`. Let me know what you think!

Reply:
I believe you're thinking that it would be useful to record the fact that this error was caught as a promise rejection event rather than an error event.

Making this distinction could be useful for debugging. However, we must apply conventions consistently across error types. The current behavior for uncaught errors does not prefix the error type with how the error was caught.

We can (and maybe should) change this behavior, but this will require some planning. E.g., do we add another field, do we add prefixes like `Uncaught TypeError`, `Uncaught (in promise) TypeError`, `Caught TypeError`, etc.

Reply:
I do agree with your point that we should be consistent with the current conventions.
In the next revision, I will add a comment in the code mentioning we should make this improvement in the future and also raise an action item.